test/h-group/level-5.js

```diff
@@ -263,6 +263,40 @@ describe('guide principle', () => {
 		const action = take_action(game);
 		ExAsserts.objHasProperties(action, { type: ACTION.RANK, target: 1, value: 5 });
 	});
+
+	it(`understands a critical save while finessed when other potential givers are finessed`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx'],
+			['p1', 'p2', 'p3', 'g3'],
+			['p4', 'y5', 'b3', 'p5'],
+			['b4', 'y2', 'p4', 'r4']
+		], { level: { min: 5 }, starting: PLAYER.CATHY, play_stacks: [0, 0, 0, 0, 0] });
+		takeTurn(game, 'Cathy clues purple to Donald'); // finesses p1, p2, p3
+		takeTurn(game, 'Donald clues blue to Cathy'); // finesses b1, b2 in our hand
+		takeTurn(game, 'Alice clues 5 to Cathy');
+
+		// Understands that Alice may have been deferring the finesse to save the 5 and allow Bob to play.
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order].finessed, true);
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order], ['b1']);
+	});
+
+	it(`understands a critical save where other players only have a play if we play`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx'],
+			['b1', 'p2', 'g4', 'g2'],
+			['y4', 'y5', 'p3', 'b5'],
+			['b4', 'y2', 'p4', 'r4']
+		], { level: { min: 5 }, starting: PLAYER.DONALD, play_stacks: [0, 0, 0, 0, 0] });
+		takeTurn(game, 'Donald clues purple to Cathy'); // finesses p1 (Alice), b1 (Bob), p2 (Bob)
+
+		// Bob may think playing gives Cathy a play, but Alice can see that it doesn't,
+		// and should save Cathy's 5.
+		takeTurn(game, 'Alice clues 5 to Cathy');
```

Comment:
Note that take_action does not currently identify this action. I think that it probably should, but figuring this out is probably going to be a bit complex. E.g. I want it to only clue the save in cases like this where we know that it won't become playable, but not delay playing into the finesse if it thinks it will become playable.

Reply:
Actually it looks a bit more complex than that, `high_priority` is false in this test case because it's early game and 5 is an early game clue, but it seems like such a clue should still be treated as `high_priority` if we don't know that Cathy will be otherwise busy.

Also in this particular test case I've set up, arguably Alice should protect Bob's 2 who should then protect Cathy's 5's, in case there's nothing left to clue after the 5 save.

Reply:
I'm also thinking we can make urgent-action symmetric with the cases that are recognized as allowed finesse deferrals by simulating the clue (it already simulates some clues in `find_urgent_actions`) and seeing if the waiting connection is dropped as a result.

Reply:
Done! I *think* this is a lot cleaner now. Instead of needing to be careful that `take_action` and `interpret_clue` agree, we now use the results of simulating the clue in `evaluate_clue` to determine if `interpret-clue` recognizes the delayed finesse such that `update_turn` no longer removes it.